src/phylum/ci/ci_base.py

```diff
@@ -0,0 +1,310 @@
+            print(" [!] The analysis is complete and there were failures")
+            returncode = ReturnCode.FAILURE
+            output = FAILED_COMMENT
+            for line in risk_data:
+                output += line
+
+        if self.gbl_incomplete:
+            incomplete_pkg_count = len(self.incomplete_pkgs)
+            print(f" [+] {incomplete_pkg_count} packages were incomplete in the analysis results")
+            if self.gbl_failed:
+                print(" [!] There were failures in one or more completed packages")
+                returncode = ReturnCode.FAILURE_INCOMPLETE
+                output = FAILED_INCOMPLETE_COMMENT_TEMPLATE.substitute(count=incomplete_pkg_count)
+                for line in risk_data:
+                    output += line
+            else:
+                print(" [+] There were no failures in the packages that have completed so far")
+                returncode = ReturnCode.INCOMPLETE
+                output = INCOMPLETE_COMMENT_TEMPLATE.substitute(count=incomplete_pkg_count)
+
+        if not self.gbl_failed and not self.gbl_incomplete:
+            print(" [+] The analysis is complete and there were NO failures")
+            returncode = ReturnCode.SUCCESS
+            output = SUCCESS_COMMENT
+
+        output += f"\n[View this project in the Phylum UI]({project_url})"
+        self._analysis_output = output
+
+        return returncode
+
+    def parse_risk_data(self, analysis_results: dict, packages: Packages) -> List[str]:
+        """Parse risk packages from a Phylum analysis.
+
+        Packages that are in a completed analysis state will be included in the risk score report.
+        Packages that have not completed analysis will be included with other incomplete packages
+        and the overall PR will be allowed to pass, but with a note about re-running again later.
+        """
+        analysis_pkgs = analysis_results.get("packages", [])
+        risk_scores = []
+        for package in packages:
+            for phylum_pkg in analysis_pkgs:
+                if phylum_pkg.get("name") == package.name and phylum_pkg.get("version") == package.version:
+                    if phylum_pkg.get("status") == "complete":
+                        risk_score = self.check_risk_scores(phylum_pkg)
+                        if risk_score:
+                            risk_scores.append(risk_score)
+                    elif phylum_pkg.get("status") == "incomplete":
+                        self.incomplete_pkgs.append(package)
+                        self.gbl_incomplete = True
+
+        return risk_scores
+
+    def check_risk_scores(self, package_result: dict) -> Optional[str]:
+        """Check risk scores of a package against user-provided thresholds.
+
+        If a package has a risk score below the threshold, set the fail flag and generate the markdown output.
+        """
+        failed_flag = False
+        risk_vectors = package_result.get("riskVectors", {})
```

Comment:
So... `riskVectors` may not be around much longer (based on [this discussion](https://phylum-dev.slack.com/archives/C037VKS83CJ/p1652221778518439))....

Reply:
Roger that...will BOLO for that.